acp/ACP.md

```diff
@@ -0,0 +1,419 @@
+
+3. **Preventing Data Breaches**: Access control acts as a proactive measure to deter, detect, and prevent unauthorized access. It ensures that only those with the necessary permissions can access sensitive data or services.
+
+4. **Managing Complexity**: Modern IT infrastructure, including cloud computing and mobile devices, has exponentially increased the number of access points. Technologies like **identity and access management (IAM)** and approaches like **zero trust** help manage this complexity effectively.
+
+## Types of Security Access Controls
+
+Several access control models exist, including:
+
+- **Role-Based Access Control (RBAC)**: Assigns permissions based on predefined roles (e.g., admin, user, manager).
+- **Attribute-Based Access Control (ABAC)**: Considers various attributes (e.g., user attributes, resource attributes) for access decisions.
+- **Discretionary Access Control (DAC)**: Allows resource owners to set access permissions.
+- **Mandatory Access Control (MAC)**: Based on security labels and clearances.
+- **Policy-Based Access Control (PBAC)**: Enforces access based on defined policies.
+- **Relation-Based Access Control (ReBac)**: Enforces access based on defined relationships within policies.
+
+- Note: **DefraDB** uses **Relation-Based Access Control System (ReBac)**
+
+## Challenges of Access Control in Cybersecurity
+
+- **Distributed IT Environments**: Cloud computing and remote work create new challenges.
+- **Rise of Mobility**: Mobile devices in the workplace add complexity.
+- **Password Fatigue**: Balancing security with usability.
+- **Data Governance**: Ensuring visibility and control.
+- **Multi-Tenancy**: Managing complex permissions in SaaS applications.
+
+## Key takeaway
+A robust access control policy system is your first line of defense against unauthorized access and data breaches.
+
+
+# DefraDB's Access Control System
+
+## ReBac Authorization Model
+
+### Zanzibar
+In 2019, Google published their [Zanzibar](https://research.google/pubs/zanzibar-googles-consistent-global-authorization-system/) paper, a paper explaining how they handle authorization across their many services. It uses access control lists but with relationship-based access control rather than role-based access control. Relationship-Based Access Control (ReBAC) establishes an authorization model where a subject's permission to access a resource is defined by the presence of relationships between those subjects and resources.
+The way Zanzibar works is that you write a tuple that includes subject, relation, and object. Then, you check if a permission exists and run an expand call that acts as a reverse lookup over a user set. It operates as a Consistent and Partition-Tolerant System.
```

Comment:
suggestion: I think the phrasing is not clear for people not familiar with Zanzibar lingo.

Zanzibar exposes an API with (mainly) operations to manage Relationships (tuples) and verify Access Requests (can Bob do X) through the Check call. The Check call performs a a Graph Search over the tuples to find a path between the user and the object, if such a path exist then according to RelBAC the user has the queried permission.


Zanzibar Checks

Reply:
- [x] Done